service/ztests/curl-cors.yaml

```
@@ -0,0 +1,51 @@
+script: |
+  source service.sh
+  echo "=== OPTIONS: allowed ==="
+  curl -sD - \
+    -X OPTIONS \
+    -H "Access-Control-Request-Method: POST" \
+    -H "Access-Control-Request-Headers: content-type" \
+    -H "Origin: http://test.observableusercontent.com" \
+    -H "Accept: application/json" \
+    -H "Content-Type: application/json" \
+    $ZED_LAKE/query | grep -Fi "Access-Control-Allow"  | tr -d '\015'
+  echo "=== OPTIONS: not allowed ==="
+  curl -sD - \
+    -X OPTIONS \
+    -H "Access-Control-Request-Method: POST" \
+    -H "Access-Control-Request-Headers: content-type" \
+    -H "Origin: http://adversarialobservableusercontent.com" \
+    -H "Accept: application/json" \
+    -H "Content-Type: application/json" \
+    $ZED_LAKE/query | grep -Fi "Access-Control-Allow"  || true
```

Use `!` instead of `|| true` in ztests.
```suggestion
  ! curl -sD - \
    -X OPTIONS \
    -H "Access-Control-Request-Method: POST" \
    -H "Access-Control-Request-Headers: content-type" \
    -H "Origin: http://adversarialobservableusercontent.com" \
    $ZED_LAKE/query | grep Access-Control-Allow
```